simplex/block.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package simplex
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/ava-labs/simplex"
+
+	"github.com/ava-labs/avalanchego/snow/consensus/snowman"
+	"github.com/ava-labs/avalanchego/snow/engine/snowman/block"
+	"github.com/ava-labs/avalanchego/utils/hashing"
+)
+
+var _ simplex.BlockDeserializer = (*blockDeserializer)(nil)
+
+var _ simplex.Block = (*Block)(nil)
+
+var _ simplex.VerifiedBlock = (*Block)(nil)
+
+var errBlockAlreadyVerified = errors.New("block has already been verified")
+
+type Block struct {
+	digest simplex.Digest
+
+	// metadata contains protocol metadata for the block
+	metadata simplex.ProtocolMetadata
+
+	verified bool // whether vmBlock.verify() function has been called
+
+	// the parsed block
+	vmBlock snowman.Block
+}
+
+// BlockHeader returns the block header for the verified block.
+func (b *Block) BlockHeader() simplex.BlockHeader {
+	return simplex.BlockHeader{
+		ProtocolMetadata: b.metadata,
+		Digest:           b.digest,
+	}
+}
+
+// Bytes returns the serialized bytes of the verified block.
+func (b *Block) Bytes() ([]byte, error) {
+	cBlock := &CanotoSimplexBlock{
+		Metadata:   b.metadata.Bytes(),
+		InnerBlock: b.vmBlock.Bytes(),
+	}
+
+	buff := cBlock.MarshalCanoto()
+
+	return buff, nil
```

Comment:
Not a meaningful diff, but it seems like we can probably skip the declaration of `buff`.
```suggestion
	return cBlock.MarshalCanoto(), nil
```